workspaces/ui-v2/src/optic-components/changelog/ChangelogDropdown.tsx

```diff
@@ -79,62 +76,57 @@ export function ChangesSinceDropdown(props: any) {
         style={{ marginTop: 20 }}
       >
         {allBatchCommits.map((i, index) => (
-          <BatchCommitMenuItem
-            batch={index === 0 ? { ...i, commitMessage: 'Latest' } : i}
-            key={index}
+          <MenuItem
+            key={i.batchId}
```

Comment:
We shouldn't actually use the index as a key - the way react works is that it needs an id that maps deterministically to an item - using the index could lead to bugs where we rerender the wrong item (i believe)